vassal-app/src/main/java/bsh/BSHUnaryExpression.java

```diff
@@ -82,14 +82,19 @@ private Object lhsUnaryOperation( LHS lhs, boolean strictJava )
 
     private Object unaryOperation( Object op, int kind ) throws UtilEvalError
     {
-        if (op instanceof Boolean || op instanceof Character 
+        // VASSAL - Special handling for ! operator. Don't thow errors
+        // Anything that is not a Boolean should return false, unless it is the string "true"
+        if (kind == ParserConstants.BANG ) {
+          return primitiveWrapperUnaryOperation((op instanceof Boolean ? op : new Boolean("true".equals(op))), kind );
```

Comment:
The `Boolean` ctors are deprecated. Instead of `new Boolean("true".equals(op))`, what you probably want is `Boolean.valueOf(op)` instead.